.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,46 @@
+name: Continuous integration
+on: [push]
+jobs:
+  backend:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:11.5-alpine
+        ports:
+        - 5433:5432
+    steps:
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6.9
+    - name: Update pip
+      run: python -m pip install --upgrade pip
+    - name: Install tox
+      run: python -m pip install tox
+    - name: Checkout git repo
+      uses: actions/checkout@v2
+    - name: Create test databases
+      run: |
+        psql -U postgres -h localhost -p 5433 -c 'CREATE DATABASE lms_test'
+        psql -U postgres -h localhost -p 5433 -c 'CREATE DATABASE lms_functests'
+        psql -U postgres -h localhost -p 5433 -c 'CREATE DATABASE lms_bddtests'
+    - name: Build frontend assets
+      run: |
+        yarn install
+        yarn build
+    - name: Run tox
+      run: tox --parallel all --parallel-live -e checkformatting,lint,tests,functests,bddtests
+  frontend:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Install
+      run: |
+        yarn install
+    - name: Format
+      run: yarn checkformatting
+    - name: Lint
+      run: yarn lint
+    - name: Test
+      run: gulp test
```

Comment:
These three run in serial because I don't know of a frontend command to run them parallel like we can do with tox for the backend checks. It doesn't really matter: these three are all really quick, and the backend job always takes much longer than the frontend job.

@robertknight Could use your feedback on whether `yarn install` is really necessary here or can something smaller be done?

Reply:
`yarn install` is correct, although in CI you should use the `--frozen-lockfile` flag per https://stackoverflow.com/a/58525708. In frontend projects using npm, you should use `npm ci` rather than `npm install`.

In Travis builds we used to configure caching of the `node_modules/` folder. It may be possible to do the same for GHA.